Author: José Manuel Santamaría Lema <panfaust@gmail.com>
Origin: https://projects.kde.org/projects/kde/kdeedu/cantor/repository/revisions/2ffe4fc9
Applied-Upstream: 4.7.3
Description: Fix detection of KAlgebra
 This patch allows Cantor to build the Kalgebra backend.
--- a/src/backends/kalgebra/CMakeLists.txt
+++ b/src/backends/kalgebra/CMakeLists.txt
@@ -1,4 +1,4 @@
-include_directories(${analitza_SOURCE_DIR}/../)
+include_directories(${ANALITZA_INCLUDES})
 
 set( KAlgebraBackend_SRCS
  kalgebrasyntaxhelpobject.cpp 
@@ -10,7 +10,7 @@ set( KAlgebraBackend_SRCS
 )
 
 kde4_add_plugin( cantor_kalgebrabackend ${KAlgebraBackend_SRCS} )
-target_link_libraries( cantor_kalgebrabackend ${KDE4_KDEUI_LIBS} cantorlibs analitza analitzagui)
+target_link_libraries( cantor_kalgebrabackend ${KDE4_KDEUI_LIBS} cantorlibs ${ANALITZA_LIBS})
 
 install( FILES kalgebrabackend.desktop  DESTINATION ${SERVICES_INSTALL_DIR}/cantor)
 install(TARGETS cantor_kalgebrabackend DESTINATION ${PLUGIN_INSTALL_DIR})
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,9 @@ include(KDE4Defaults)
 
 set(CMAKE_MODULE_PATH ${cantor_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
 
+macro_optional_find_package(Analitza)
+macro_log_feature(ANALITZA_FOUND "libanalitza" "A library provided by KAlgebra" FALSE "" "Backend to use KAlgebra with Cantor.")
+
 macro_optional_find_package(R)
 macro_log_feature(R_FOUND "R" "A free software environment for statistical computing and graphics" "http://www.r-project.org/" FALSE "" "Backend to use R with Cantor.")
 
--- a/src/backends/CMakeLists.txt
+++ b/src/backends/CMakeLists.txt
@@ -2,9 +2,9 @@ add_subdirectory(null)
 add_subdirectory(maxima)
 add_subdirectory(octave)
 
-if(BUILD_kalgebra)
+if(ANALITZA_FOUND)
 	add_subdirectory(kalgebra)
-endif(BUILD_kalgebra)
+endif(ANALITZA_FOUND)
 
 if(NOT WIN32)
 	add_subdirectory(sage)
